controllers/status.go

```diff
@@ -70,6 +70,7 @@ func (r *EtcdadmClusterReconciler) updateStatus(ctx context.Context, ec *etcdv1.
 	// etcd ready when all machines have address set
 	ec.Status.Ready = true
 	ec.Status.Endpoints = strings.Join(endpoints, ",")
+	conditions.MarkTrue(ec, etcdv1.EtcdEndpointsAvailable)
```

Comment:
Adding this is useful either way, so not opposed to it, since it signifies something different, something even closer to the cluster being ready than the machines being ready.

But, for my own understanding, why didn't `EtcdMachinesReadyCondition` prevent this issue from happening? I understand it's a different thing that what you are proposing but, it seems like the issue was that the machines took a bit to become ready, not necessarily that etcd takes long to become ready once the machines are (this should be provider independent).

Reply:
@g-gaston that's what I checked first too. Machine's Ready condition is an aggregate of BootstrapReady and InfrastructureReady conditions. BootstrapReady becomes true when the secret containing bootstrap data is ready. And InfrastructureReady becomes true when the InfraMachine's ready field becomes true. In case of CAPT we saw that TinkerBell machine became ready as soon as it had an IP address. So neither of these conditions waited long enough for the Machine to have an etcd member running.

Reply:
summarizing the potential changes @g-gaston and I discussed:
Currently the EtcdMachinesReady condition is an aggregate of the Ready conditions of all existing machines.  Consider making this an aggregate of all machines the cluster is supposed to own. (Need to test this out before deciding to make the changes)
The status.Ready field gets set to tru as soon as etcd members pass healthcheck. Determine if it makes more sense to set this field based on the value of the Ready condition instead.

Reply:
opened an issue to track this https://github.com/aws/eks-anywhere/issues/1383